accepted/future-releases/constructor-tearoffs/feature-specification.md

```diff
@@ -355,8 +355,9 @@ For an expression of the form <code>*e*\<*typeArgs*></code>, which is not follow
 * If *e* denotes a generic class, mixin or type alias declaration (which means that *e* is an identifier, possibly a qualified identifier, which resolves to the class, mixin or type alias declaration), then <code>*e*\<*typeArgs*></code> is a type literal. _If followed by <code>.*id*</code> then that *id* must denote a constructor, which can then be either torn off or invoked. If followed by `==` or `!=` or any "stop-token", the expression evaluates to a `Type` object._
 * If *e* denotes a generic top-level, static or local function declaration (again *e* is an identifier or qualified identifier), that declaration must be a generic function declaration, and then <code>*e*\<*typeArgs*></code> performs an explicitly instantiated function tear-off, which works just like the current implicitly instantiated function tear-off except that the types are provided instead of inferred.
 * If *e* denotes a generic instance method (*e* has the form <code>*r*.*name*</code> and *r* has a static type for which *name* is a generic interface method), then <code>*e*\<*typeArgs*></code> performs an explicitly instantiated method tear-off, which works just like the current implicitly instantiated method tear-off except that the types are provided instead of inferred.
-* If *e* has a static type which is a generic callable object type (a non-function type with a generic method named `call`), then <code>*e*\<*typeArgs*></code> is equivalent to the instantiated method-tear off <code>*e*\.call<*typeArgs*></code>.
-* Otherwise, if *e* has a static type which is a generic function type, then <code>*e*\<*typeArgs*></code> is equivalent to the instantiated method-tear off <code>*e*\.call<*typeArgs*></code>.
+* If *e* has a static type which is a generic callable object type (an interface type with a generic method named `call`), then <code>*e*\<*typeArgs*></code> is equivalent to the instantiated method tear-off <code>*e*.call\<*typeArgs*></code>.
+* If *e* has an accessible and applicable generic extension method named `call` then <code>*e*\<*typeArgs*></code> is equivalent to the instantiated extension method tearoff <code>*e*.call\<*typeArgs*></code>.
```

Comment:
I guess that's OK now that we can instantiate anything. If you can tear off `e.call`, you can also instantiate it. (IIRC, we don't do implicit extension-`.call` tear-offs, but the type arguments make this explicit. So `Function f = e;` is invalid but `Function f = e<int>;` is valid.).

Reply:
Right, we did make it an error to tear off an extension method named `call` implicitly. E.g., `Function f = e;` where `e` has an interface type with a method named `call` is an error, it does not desugar to `Function f = e.call;`, presumably because there is no syntactic hint at all on `e` about the special desugaring.

It does not appear to be overwhelmingly consistent, but we do have a rule which says that a `.call` desugaring step is always marked by explicit syntax.